### PR TITLE
test for slowness, and better way to set layer

### PR DESCRIPTION
**Pull request by doutriaux1**
@danlipsa what about this, it also sets the layer in a consistent way.

### CODE REVIEW

**Comment by danlipsa** [comment]
@doutriaux1 I wonder if you should do a line fitting instead instead of testing for time be less than 150%.
https://docs.scipy.org/doc/numpy/reference/generated/numpy.polyfit.html
You should get a horizontal line. For y = a * x + b, a has to be very small.

**Comment by doutriaux1** [comment]
@danlipsa good idea I'll do that.

**Comment by danlipsa** [comment]
@doutriaux1 I think you should scale your times so that the fastest time will become 1. So instead of "times[5:]" you should have "times[5:] / fastest"

This way, your slope of 0.001 would mean 10% of the fastest time (because you have 100 times).

File: vcs/utils.py [hunk 1]
```diff
@@ -2256,6 +2256,8 @@ def creategraphicsmethod(gtype, gname='default', name=None):
         func = vcs.create1d
     elif gtype in ['vector', 'Gv']:
         func = vcs.createvector
+    elif gtype in ['streamline', 'Gs']:
+        func = vcs.createstreamline
```
Comment (aashish24) [inline vcs/utils.py]: was this intended?

File: vcs/vcsvtk/pipeline1d.py [hunk 2]
```diff
@@ -103,7 +104,7 @@ def plot(self, data1, data2, tmpl, grid, transform, **kargs):
                 self._context().canvas.plot(m, donotstoredisplay=True)
 
         ren2 = self._context().createRenderer()
-        ren2.SetLayer(1)
+        self._context().setLayer(ren2, l.priority)
```
Comment (aashish24) [inline vcs/vcsvtk/pipeline1d.py]: didn't @danlipsa has this change in his branch? may be merge his branch into yours?

Comment (danlipsa) [inline vcs/vcsvtk/pipeline1d.py]: @aashish24 I'll close my branch, @doutriaux1 told me he is going to fix this in his branch.

File: vcs/utils.py (hunk 1 above)
Comment (doutriaux1) [inline vcs/utils.py]: @danlipsa had forgotten it in his streamline branch. When i ran my test on streamline it couldn't generate the gm so I fixed it here

Comment (aashish24) [inline vcs/utils.py]: okay, thanks @doutriaux1

File: vcs/vcsvtk/pipeline1d.py (hunk 2 above)
Comment (aashish24) [inline vcs/vcsvtk/pipeline1d.py]: thank you @danlipsa

**Comment by aashish24** [comment]
@doutriaux1 @danlipsa looks like tests are failing ? 

	tests/test_vcs_basic_gms_2.py
	tests/test_vcs_no_slowdown.py
	tests/test_vcs_vectors_robinson.py

**Comment by doutriaux1** [comment]
@danlipsa this should trigger the nightlies

**Pull request merged**